modules/javafx.graphics/src/main/java/javafx/scene/Scene.java

```diff
@@ -1301,11 +1298,55 @@ static WritableImage doSnapshot(Scene scene,
             height = (int)wimg.getHeight();
         }
 
+        int maxTextureSize = PrismSettings.maxTextureSize;
+        if (height > maxTextureSize || width > maxTextureSize) {
+            // The requested size for the screenshot is too big to fit a single texture,
+            // so we need to take several snapshot tiles and merge them into single image (fixes JDK-8088198)
+            int numVerticalTiles = (int) Math.ceil(height / (double) maxTextureSize);
+            int numHorizontalTiles = (int) Math.ceil(width / (double) maxTextureSize);
+            for (int i = 0; i < numHorizontalTiles; i++) {
+                int xOffset = i * maxTextureSize;
+                int tileWidth = Math.min(maxTextureSize, width - xOffset);
+                for (int j = 0; j < numVerticalTiles; j++) {
+                    int yOffset = j * maxTextureSize;
+                    int tileHeight = Math.min(maxTextureSize, height - yOffset);
+                    WritableImage tile = doSnapshotTile(scene, xMin + xOffset, yMin + yOffset, tileWidth, tileHeight, root, transform, depthBuffer, fill, camera, null);
+                    wimg.getPixelWriter().setPixels(xOffset, yOffset, tileWidth, tileHeight, tile.getPixelReader(), 0, 0);
+                }
+            }
```

Comment:
I'm not 100% convinced this would really add much to the readability of the code; I extracted the code from `doSnapshotTile` in its own method because it is called twice (on both sides of the `if (height > maxTextureSize || width > maxTextureSize)` condition, actually), but this isn't the case here.
I've got no strong feeling against it either, so I don't know; anybody else care to comment?

Reply:
I also don't have a strong opinion, so I'm OK with you leaving it as-is.

Reply:
Fine to leave as-is.